x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/reroute.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiCode } from '@elastic/eui';
+
+import { get } from 'lodash';
+import {
+  ComboBoxField,
+  FIELD_TYPES,
+  UseField,
+  Field,
+  useFormContext,
+} from '../../../../../../shared_imports';
+
+import { FieldsConfig, to, from } from './shared';
+
+const fieldsConfig: FieldsConfig = {
+  /* Optional field configs */
+  destination: {
+    type: FIELD_TYPES.TEXT,
+    serializer: from.emptyStringToUndefined,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.reroute.destinationFieldLabel', {
+      defaultMessage: 'Destination (optional)',
+    }),
+    helpText: (
+      <FormattedMessage
+        id="xpack.ingestPipelines.pipelineEditor.reroute.destinationFieldHelperText"
+        defaultMessage="A static value for the target. Cannot be set when {dataset} or {namespace} is set."
+        values={{
+          dataset: <EuiCode>{'dataset'}</EuiCode>,
+          namespace: <EuiCode>{'namespace'}</EuiCode>,
+        }}
+      />
+    ),
+  },
+  dataset: {
+    defaultValue: null,
+    type: FIELD_TYPES.COMBO_BOX,
+    deserializer: to.arrayOfStrings,
+    serializer: from.optionalArrayOfStrings,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.reroute.datasetFieldLabel', {
+      defaultMessage: 'Dataset (optional)',
+    }),
+    helpText: (
+      <FormattedMessage
+        id="xpack.ingestPipelines.pipelineEditor.reroute.datasetFieldHelperText"
+        defaultMessage="Field references or a static value for the dataset part of the data stream name. In addition to the criteria for index names, cannot contain {dash} and must be no longer than 100 characters. Defaults to {defaultValue}."
```

Comment:
nit: should this be `<it> cannot contain dash..` ? I see that in the docs is written without it also, but sounds strange? 🤔 

```suggestion
        defaultMessage="Field references or a static value for the dataset part of the data stream name. In addition to the criteria for index names, it cannot contain {dash} and must be no longer than 100 characters. Defaults to {defaultValue}."
```